src/main/java/teammates/sqllogic/core/DeadlineExtensionsLogic.java

```diff
@@ -72,4 +77,35 @@ public DeadlineExtension createDeadlineExtension(DeadlineExtension deadlineExten
         assert deadlineExtension != null;
         return deadlineExtensionsDb.createDeadlineExtension(deadlineExtension);
     }
+
+    /**
+     * Deletes a deadline extension.
+     *
+     * <p>Fails silently if the deadline extension doesn't exist.</p>
+     */
+    public void deleteDeadlineExtension(DeadlineExtension deadlineExtension) {
+        deadlineExtensionsDb.deleteDeadlineExtension(deadlineExtension);
+    }
+
+    /**
+     * Deletes the user email address for all their deadlines in the feedback sessions of the given course.
+     */
+    public void deleteFeedbackSessionsDeadlinesForUser(String courseId, String emailAddress) {
+        List<FeedbackSession> feedbackSessions = feedbackSessionsLogic.getFeedbackSessionsForCourse(courseId);
+
+        feedbackSessions.forEach(feedbackSession -> {
+            List<DeadlineExtension> deadlineExtensions = feedbackSession.getDeadlineExtensions();
+
+            deadlineExtensions = deadlineExtensions
+                    .stream()
+                    .filter(deadlineExtension ->
+                            deadlineExtension.getUser().getEmail().equalsIgnoreCase(emailAddress))
+                    .collect(Collectors.toList());
+
+            for (DeadlineExtension deadlineExtension : deadlineExtensions) {
+                deleteDeadlineExtension(deadlineExtension);
+            }
+        });
+    }
```

Comment:
This should be `deleteDeadlineExtensions for user`. Parameters should just be User user.

The bug here is that the email does not differentiate between instructor and student.